pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -1008,14 +1011,34 @@ func (r *Builder) ensureVolumePopulatorPVC(workload *model.Workload, image *mode
 			originalVolumeDiskId = imageProperty.(string)
 		}
 
-		storageClassName := r.Context.Map.Storage.Spec.Map[0].Destination.StorageClass
-		if volumeType := r.getVolumeType(workload, originalVolumeDiskId); volumeType != "" {
-			storageClassName, err = r.getStorageClassName(workload, volumeType)
-			if err != nil {
-				err = liberr.Wrap(err)
-				return
+		mapList := r.Context.Map.Storage.Spec.Map
+
+		// Check if there's a storage map available
+		if len(mapList) == 0 {
+			err = liberr.New("no storage map found in the migration plan")
+			return
+		}
+
+		var storageClassName string
+
+		// VM is image based, look for a glance key in the mapping
+		if workload.ImageID != "" {
+			for _, storageMap := range mapList {
+				if storageMap.Source.Name == "glance" {
+					storageClassName = storageMap.Destination.StorageClass
+				}
+			}
+		} else {
+			// VM is volume based, look for the volume type in the mapping
+			if volumeType := r.getVolumeType(workload, originalVolumeDiskId); volumeType != "" {
+				storageClassName, err = r.getStorageClassName(workload, volumeType)
+				if err != nil {
+					err = liberr.Wrap(err)
```

Comment:
I'm not sure why we need this check (shouldn't we always a corresponding storage class since otherwise we would have set the storage mapping as invalid?) but since we have it here, do we need it for glance above as well?

Reply:
I think it's just there, the mapping has to be valid at this point

Reply:
so please add a comment that reflects this